lib/routes.js

```diff
@@ -1,17 +1,20 @@
 module.exports = (self, options) => {
   self.addRoutes = function() {
     self.route('get', 'render/:id', async (req, res) => {
-      const box = await self.findOne(req);
+      try {
+        const box = await self.findOne(req);
+        if (box) {
+          return res.status(200).send(self.render(
+            req,
+            `apostrophe-dialog-box:dialogs/${box.template}.html`,
+            { piece: box }
+          ));
+        }
 
-      if (box) {
-        return res.status(200).send(self.render(
-          req,
-          `apostrophe-dialog-box:dialogs/${box.template}.html`,
-          { piece: box }
-        ));
+        return res.status(404).send('');
+      } catch (error) {
+        console.log(error);
```

Comment:
You already caught the error in findOne so you won't get here. Recommend you remove the catch in findOne and use this one catch to do the right thing for all of the error cases.

Reply:
👁